packages/twenty-server/src/engine/metadata-modules/remote-server/remote-server.service.ts

```diff
@@ -0,0 +1,101 @@
+import { Injectable, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+
+import { TypeOrmQueryService } from '@ptc-org/nestjs-query-typeorm';
+import { Repository } from 'typeorm';
+
+import { TypeORMService } from 'src/database/typeorm/typeorm.service';
+import { CreateRemoteServerInput } from 'src/engine/metadata-modules/remote-server/dtos/create-remote-server.input';
+import {
+  RemoteServerEntity,
+  RemoteServerType,
+} from 'src/engine/metadata-modules/remote-server/remote-server.entity';
+import { EnvironmentService } from 'src/engine/integrations/environment/environment.service';
+import { encryptText } from 'src/engine/core-modules/auth/auth.util';
+
+@Injectable()
+export class RemoteServerService extends TypeOrmQueryService<RemoteServerEntity> {
+  constructor(
+    @InjectRepository(RemoteServerEntity, 'metadata')
+    private readonly remoteServerRepository: Repository<RemoteServerEntity>,
+    private readonly typeORMService: TypeORMService,
+    private readonly environmentService: EnvironmentService,
+  ) {
+    super(remoteServerRepository);
+  }
+
+  async createOneRemoteServer(
+    remoteServerInput: CreateRemoteServerInput,
+    workspaceId: string,
+  ): Promise<RemoteServerEntity> {
+    const mainDatasource = this.typeORMService.getMainDataSource();
+    const key = this.environmentService.get('LOGIN_TOKEN_SECRET');
+    const iv = this.environmentService.get('IV_SECRET');
+    const encryptedPassword = await encryptText(
+      remoteServerInput.password,
+      key,
+      iv,
+    );
+
+    const createdRemoteServer = await super.createOne({
+      ...remoteServerInput,
+      password: encryptedPassword,
+      workspaceId,
+    });
+
+    await mainDatasource.query(
+      `CREATE SERVER IF NOT EXISTS "${createdRemoteServer.fdwId}" FOREIGN DATA WRAPPER postgres_fdw OPTIONS (dbname '${remoteServerInput.database}', host '${remoteServerInput.host}', port '${remoteServerInput.port}')`,
```

Comment:
this will not work for all wraper types, some wrapers don't have a dbname or a host/port, it can be just an api key

Reply:
I wanted it working only for postgres so fare, but I can definitely switch on type already